buildSrc/src/main/kotlin/otel.java-conventions.gradle.kts

```diff
@@ -237,7 +237,6 @@ testing {
       compileOnly("com.google.auto.value:auto-value-annotations")
       compileOnly("com.google.errorprone:error_prone_annotations")
       compileOnly("com.google.code.findbugs:jsr305")
-      compileOnly("com.google.code.findbugs:annotations")
```

Comment:
Reverting https://github.com/open-telemetry/opentelemetry-java/pull/6912/commits/c20dfe1b6785aa4c0af72dbf114a1e0150b41977 now that https://github.com/jqno/equalsverifier/issues/1026 is fixed.